src/foolscap/schema.py

```diff
@@ -133,12 +134,17 @@ def AnyStringConstraint(*args, **kwargs):
 # AnyStringConstraint
 StringConstraint = ByteStringConstraint
 
+maxBytesMap = {
+    int: -1,
+    long: 1024
+    }
+
 constraintMap = {
-    str: ByteStringConstraint(),
+    six.binary_type: ByteStringConstraint(),
     unicode: UnicodeConstraint(),
     bool: BooleanConstraint(),
     int: IntegerConstraint(),
-    long: IntegerConstraint(maxBytes=1024),
+    long: IntegerConstraint(maxBytes=maxBytesMap[long]),
```

Comment:
Okay, but.. why? It's plausible that this is the correct thing to do, but `-1` isn't in the original code, so this looks like changing behavior.